recipe/meta.yaml

```
@@ -36,13 +36,12 @@ outputs:
       host:
         - fmt
         - libabseil
+        - libboost-devel
```

It seems that obake does not need any of the shared libraries from boost at runtime

```suggestion
        - libboost-headers
```

At least according to the link check:
```
WARNING (obake): dso library package conda-forge::libboost-1.82.0-h6fcfa73_6 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
```

---

Hi @h-vetinari and thanks a lot for the help!

So obake is *mostly* header-only, but it has some compiled parts. Whether or not symbols from the Boost libraries end up being used in the obake shared library is going to be mostly an implementation detail potentially hard to pin down precisely and subject to change over time.

Under such circumstances, how should one proceed? Personally, I would prefer to potentially overdepend rather than the alternative. I understand that ``conda-build`` would give me a warning/error if the obake SO ends up depending on Boost and Boost is not listed as a dependency, but this would mean realising the mistake only when the binary package is being built.

---

> Whether or not symbols from the Boost libraries end up being used in the obake shared library is going to be mostly an implementation detail potentially hard to pin down precisely and subject to change over time.

That's what the link-check is supposed to validate.



> Under such circumstances, how should one proceed?

I would recommend relying just on the headers now, but also adding
```
conda_build:
  error_overlinking: true
```
under `conda-forge.yml`. That way, if you ever rely on compiled bits of boost (without reflecting it in the metadata), you'll get a build error.

---

Am I reading this right that this is ``overlinking`` and not ``underlinking`` or something like that?

---

"overlinking" is when your library is linking to a dependency without having declared it in host. In other words, if you start depending on having some of the `libboost_*.so` binaries available implicitly (e.g. they're in the environment as a transitive dependency; otherwise things would already fail during the build directly), then the build would error because the link check figures out that you're loading symbols from a SO that's not in any of the outputs on which you have an explicit run-dependence.